lib/services/elasticsearch.js

```diff
@@ -73,6 +75,15 @@ class ElasticSearch extends Service {
       return Bluebird.resolve(this);
     }
 
+    this.config.dynamicMapping = this.config.dynamicMapping.toString();
+    if (this.dynamicMappingPolicies.indexOf(this.config.dynamicMapping) === -1) {
+      throw new KuzzleInternalError(`Wrong configuration value '${this.config.dynamicMapping}' for server.db.dynamicMapping. Allowed values are the following string '${this.dynamicMappingPolicies.join('\', \'')}'.`);
+    }
+
+    if (process.env.NODE_ENV === 'production' && this.config.dynamicMapping !== 'strict') {
+      this.kuzzle.pluginsManager.trigger('log:warn', 'You are not using the \'strict\' policy for collection mappings.\nYou should strictly define your mapping before sending data to prevent Elasticsearch default mapping types.');
```

Comment:
(nitpicking) Use a string litteral so you don't have to escape single quotes.

Reply:
Eslint is not happy when I use string litteral without variable interpolation :(

Reply:
I checked the `quotes` eslint rule configuration, and there is no easy way to configure it like this: "single quotes, allow backticks if it prevents escapes or if it uses interpolation"

There is a "avoidEscapes" configuration, but that only allows (in our case) using double quotes if it  avoids escaping single quotes.

So, ok :+1: